lib/Inst/Inst.cpp

```diff
@@ -241,6 +248,25 @@ void ReplacementContext::setBlock(llvm::StringRef Name, Block *B) {
   BlockNames[B] = Name;
 }
 
+std::string Inst::getKnownBitsPattern(llvm::APInt Zero, llvm::APInt One) {
+  std::string Str;
+  for (unsigned K=Zero.getBitWidth(); K>0; --K) {
+    if (Zero.getHiBits(1) == 1 && One.getHiBits(1) == 1)
+      llvm_unreachable("KnownZero and KnownOnes bit can't be set to 1 together");
+    if (Zero.getHiBits(1) == 1) {
+      Str.append("0");
+    } else {
+      if (One.getHiBits(1) == 1)
+        Str.append("1");
+      else
+        Str.append("x");
+    }
+    One = One.shl(1);
+    Zero = Zero.shl(1);
+  }
```

Comment:
there's a better way to do this: just use the overloaded array operator to get the individual bits. no shifting needed.